common.mk

```diff
@@ -4,13 +4,43 @@
 SHELL=/bin/bash
 
 #########################################################################################
-# extra make variables/rules from subprojects
-#
-# EXTRA_GENERATOR_REQS - requirements needed for the main generator
-# EXTRA_SIM_FLAGS - runtime simulation flags
-# EXTRA_SIM_CC_FLAGS - cc flags for simulators
-# EXTRA_SIM_SOURCES - simulation sources needed for simulator
-# EXTRA_SIM_REQS - requirements to build the simulator
+# specify user-interface variables
+#########################################################################################
+HELP_COMPILATION_VARIABLES += \
+"   EXTRA_GENERATOR_REQS   = requirements needed for the main generator" \
+"   EXTRA_SIM_CFLAGS       = CFLAGS for building simulators" \
+"   EXTRA_SIM_CXXFLAGS     = CXXFLAGS for building simulators" \
+"   EXTRA_SIM_LDFLAGS      = LDFLAGS for building simulators" \
+"   EXTRA_SIM_SOURCES      = simulation sources needed for simulator" \
+"   EXTRA_SIM_REQS         = requirements to build the simulator"
+
+EXTRA_GENERATOR_REQS ?=
+EXTRA_SIM_CXXFLAGS   ?=
+EXTRA_SIM_CFLAGS     ?=
+EXTRA_SIM_LDFLAGS    ?=
+EXTRA_SIM_SOURCES    ?=
+EXTRA_SIM_REQS       ?=
+
+#----------------------------------------------------------------------------
+HELP_SIMULATION_VARIABLES += \
+"   EXTRA_SIM_FLAGS        = runtime simulation flags (passed within +permissive)" \
+"   NUMACTL                = set to '1' to wrap simulator in the appropriate numactl command"
+
+EXTRA_SIM_FLAGS ?=
+NUMACTL         ?= 0
+
+NUMA_PREFIX = $(if $(filter $(NUMACTL),0),,$(shell numa_prefix))
+
+#----------------------------------------------------------------------------
+HELP_COMMANDS += \
+"   run-binary             = run [./$(shell basename $(sim))] and output instructions" \
+"   run-binary-fast        = run [./$(shell basename $(sim))] and don't output instructions" \
+"   run-binary-debug       = run [./$(shell basename $(sim_debug))] and output instructions and waveform" \
+"   run-binary-debug-fast  = run [./$(shell basename $(sim_debug))] and don't output instructions or waveform" \
+"   verilog                = generate intermediate verilogs from chisel elaboration and firrtl passes"
```

Comment:
This list is incomplete, which might actually do more harm than good. It's also probably important to note that there are Make targets that come from the included `.d` files (like the `run-*-tests` targets).

Reply:
What's the point of `run-binary-debug-fast`?

Reply:
i'm not trying to list all 1000 make targets, just the ones that non-power-users find useful. i can add the `run-*-tests`. if any other generally useful targets are missing, please enumerate them.

Reply:
i use `run-binary-debug-fast` all the time to run the debug binary without generating the waveform or instruction trace.

Reply:
I don't think `run-binary-debug-fast` makes sense. It should just be `run-binary-fast` if you don't want the waveform or instruction trace.